src/garage/sampler/task_embedding_worker.py

```diff
@@ -0,0 +1,204 @@
+"""Worker class for task embedding."""
+import abc
+from collections import defaultdict
+
+import gym
+import numpy as np
+
+from garage import TaskEmbeddingTrajectoryBatch
+from garage.experiment import deterministic
+from garage.sampler.worker import Worker
+from garage.sampler.env_update import EnvUpdate
+
+
+class TaskEmbeddingWorker(Worker):
```

Comment:
It's probably best to inherit from `DefaultWorker` here, instead of copying the bodies of the `worker_init`, `update_agent`, and `update_env` methods.